ide/rust/sync_info_aspect.bzl

```diff
@@ -29,8 +29,11 @@ def _should_generate_dep_info(dependency):
 def _should_generate_build_dep_info(dependency):
     return dependency.kind in _TARGET_TYPES and _TARGET_TYPES[dependency.kind] == "build"
 
-def _is_universe_crate(rust_target):
-    return str(rust_target.label).startswith("@crates__")
+def _is_universe_crate(target):
+    return str(target.label).startswith("@crates__")
+
+def _universe_crate_name(target):
+    return str(target.label).split(".")[0].rsplit("-", 1)[0].removeprefix("@crates__")
```

Comment:
As per (https://semver.org/#backusnaur-form-grammar-for-valid-semver-versions), this should comply with any crate name where the name is proceeded by a valid semantic version.